internal/workflow.go

```diff
@@ -472,6 +472,26 @@ type (
 		VersioningBehavior VersioningBehavior
 	}
 
+	// DynamicRegisterOptions consists of options for registering a dynamic workflow
+	//
+	// Exposed as: [go.temporal.io/sdk/workflow.DynamicRegisterOptions], [go.temporal.io/sdk/activity.DynamicRegisterOptions]
+	DynamicRegisterOptions struct {
+		// Allows dynamic options to be loaded for a workflow.
+		LoadDynamicOptions func(name string) (DynamicWorkflowOptions, error)
```

Comment:
```suggestion
		LoadDynamicOptions func(LoadDynamicOptionsDetails) (DynamicWorkflowOptions, error)
```

Need to be more future proof on the param IMO